_posts/2020-08-20-review-apps-post.markdown

```diff
@@ -0,0 +1,308 @@
+
+For example, to feel safe deploying [a major version upgrade of Artsy's design
+system in Force][styled-systems-upgrade-pr] the most pragmatic way forward was
+to deploy that PR to a server where other engineers could collaborate on QA.
+
+*If a single shared staging environment is the only non-production server to
+deploy to, the chances that work lands on staging in an unsafe state is high*. While
+staging being unsafe isn't _itself_ a bad thing, many bad things can result:
+
+1. [Bad] Blocked Deploys
+
+	If staging is unsafe and this dangerous state is discovered, then top priority
+is getting a service back into a safe state. While working to get a service
+back to a healthy state, new features can't be delivered.
+
+	In aggregate, this can be a sizeable productivity loss.
+
+2. [Worse] Unsafe Deploys
+
+	If staging in unsafe and this dangerous state is not discovered before a production
+deploy (for example, the unsafe code might be from another team, as described above),
+then end-users might interact with a service that just isn't working. No good.
+
+3. [Terrible] Fear
+
+	> Fear is the mind-killer.
+	[Dune][dune-fear-quote]
+
+	Alright, a bit over the top, but the risk of unsafe or blocked deploys can
+implicitly or explicitly result in teams shying away from complex work.
+
+	This avoidable fear might result in increased technical debt or not taking on
+certain projects.
+
+	It's generally bad for business and does not lead to a pleasant work environment!
+
+## Problem Recap & Review App Introduction
+
+In review, when working on complex tasks, particularly on shared services, it is
+really valuable to be able to share work easily, but when you can only share your
+work on a shared staging environment, many negative outcomes can result.
+
+Review apps are simply other staging environments that are easy to spin up and
+are deployed with the version of the service that you are working on.
+
+By QA-ing on a review app instead of a shared staging environment, teams can
+take their time ensuring the safety of their changes, removing the risks
+detailed above.
+
+Larger infrastructure upgrades (including upgrades to the
+underlying K8s configuration, in addition to app-level changes) can sit on a
+server for hours or days, allowing any slow-moving issue to show itself in a
+lower risk environment.
+
+Artsy has iterated on its review app tooling to the point where Team A and Team B
+can each deploy their changes to isolated servers of our main website, Force,
+on a single `git push` to a branch matching a naming convention.
+
+The rest of this post describes Artsy's evolution of its review app tooling
+and areas for continued improvement.
+
+## Review App Tooling
+
+### Heroku Days
+
+In the beginning, Artsy deployed most services on Heroku.
+
+[Heroku review apps][heroku-review-app-docs] were used on some teams sparingly.
+
+### Enter Kubernetes and Hokusai
+
+For many reasons outside of the scope of this post, Artsy began migrating
```

Comment:
Here or elsewhere might be a good opportunity to drop in a link to other pertinent Artsy blog posts:

- [Kubernetes and Hokusai](https://artsy.github.io/blog/2019/10/18/kubernetes-and-hokusai/)
- [In the 'Whelp!' of the Great Wave](https://artsy.github.io/blog/2018/01/24/kubernetes-and-hokusai/)